src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java

```diff
@@ -670,16 +671,15 @@ private Server checkForServer(Server server, String id, String registry, String
 
     private String decrypt(String password) throws MojoExecutionException {
         try {
-            // Done by reflection since I have classloader issues otherwise
-            Object secDispatcher = container.lookup(SecDispatcher.ROLE, "maven");
-            Method method = secDispatcher.getClass().getMethod("decrypt",String.class);
-            synchronized(secDispatcher) {
-                return (String) method.invoke(secDispatcher, password);
-            }
+            SettingsDecrypter settingsDecrypter = container.lookup(SettingsDecrypter.class);
+            Server stub = new Server();
+            stub.setUsername("whatever");
+            stub.setPassword(password);
+
+            SettingsDecryptionResult result = settingsDecrypter.decrypt(new DefaultSettingsDecryptionRequest(stub));
+            return result.getServers().get(0).getPassword();
```

Comment:
What if there are multiple server credentials in `settings.xml`, e.g. against `docker.io` for the base image and e.g. `ghcr.io` for the derivated application image? And why is the username irrelevant? 
AFAIK the first a docker client will do is to login via basic auth (maybe by using a registry specific token) and retrieve a scoped bearer token (https://docs.docker.com/registry/#authentication) for further communication.